src/components/TagPicker.tsx

```diff
@@ -0,0 +1,133 @@
+import {useMemo, useState} from 'react';
+import type {OnyxCollection} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import type SafeAreaInsetProps from '@pages/SafeAreaInsetProps';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PolicyTags, RecentlyUsedTags} from '@src/types/onyx';
+import OptionsSelector from './OptionsSelector';
+
+type TagPickerOnyxProps = {
+    /** Collection of tags attached to a policy */
+    policyTags: OnyxCollection<PolicyTags>;
+
+    /** List of recently used tags */
+    policyRecentlyUsedTags: OnyxCollection<RecentlyUsedTags>;
+
+    /** Should show the selected option that is disabled? */
+    shouldShowDisabledAndSelectedOption?: boolean;
+};
+
+type TagPickerProps = TagPickerOnyxProps & {
+    /** The policyID we are getting tags for */
+    // eslint-disable-next-line react/no-unused-prop-types
+    policyID: string;
+
+    /** The selected tag of the money request */
+    selectedTag: string;
+
+    /** The name of tag list we are getting tags for */
+    tag: string;
+
+    /** Callback to submit the selected tag */
+    onSubmit: () => void;
+
+    /**
+     * Safe area insets required for reflecting the portion of the view,
+     * that is not covered by navigation bars, tab bars, toolbars, and other ancestor views.
+     */
+    insets: SafeAreaInsetProps;
```

Comment:
Use `EdgeInsets` type instead
```suggestion
    insets: EdgeInsets;
```